code/modules/projectiles/gun.dm

```diff
@@ -333,7 +363,12 @@
 	accuracy = initial(accuracy)	//Reset the gun's accuracy
 
 	if(muzzle_flash)
-		set_light(0)
+		//VOREStation Edit - Flashlights
+		if(gun_light)

```

Comment:
Lasers already do all have lights. I doubt it'll be much of an issue on the new lighting system. We have /tg/'s lights, and this gun with a light is from /tg/, so I think it'll work. In fact you can attach a maglight to tons of /tg/ guns, and they have a generic system for adding the light sprite without needing an additional sprite for every gun.